File: eHour-service/src/main/java/net/rrm/ehour/timesheet/dao/TimesheetCommentDAO.java
/**
 * Created on Dec 28, 2006
 * Created by Thies Edeling
 * Copyright (C) 2005, 2006 te-con, All Rights Reserved.
 *
 * This Software is copyright TE-CON 2007. This Software is not open source by definition. The source of the Software is available for educational purposes.
 * TE-CON holds all the ownership rights on the Software.
 * TE-CON freely grants the right to use the Software. Any reproduction or modification of this Software, whether for commercial use or open source,
 * is subject to obtaining the prior express authorization of TE-CON.
 * thies@te-con.nl
 * TE-CON
 * Legmeerstraat 4-2h, 1058ND, AMSTERDAM, The Netherlands
 *
 */

package net.rrm.ehour.timesheet.dao;

import net.rrm.ehour.dao.GenericDAO;
import net.rrm.ehour.timesheet.domain.TimesheetComment;
import net.rrm.ehour.timesheet.domain.TimesheetCommentId;

/**
 * CRUD on timesheetComment domain obj
 **/

public interface TimesheetCommentDAO  extends GenericDAO<TimesheetComment, TimesheetCommentId>
{
	/**
	 * Delete comments for user
	 * @param user
	 * @return
	 */
	public int deleteCommentsForUser(Integer userId);
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/timesheet/dao/TimesheetCommentDao.scala
package net.rrm.ehour.persistence.timesheet.dao

import java.util

import net.rrm.ehour.data.DateRange
import net.rrm.ehour.domain.{TimesheetComment, TimesheetCommentId}
import net.rrm.ehour.persistence.dao.GenericDao

/**
 * CRUD on timesheetComment domain obj
 **/
trait TimesheetCommentDao extends GenericDao[TimesheetCommentId, TimesheetComment] {
  /**
   * Delete comments for user
   */
  def deleteCommentsForUser(userId: Integer): Int

  def findCommentBetween(dateRange: DateRange): util.List[TimesheetComment]

  def findCommentBetweenForUsers(users: util.List[Integer], dateRange: DateRange): util.List[TimesheetComment]
}
